sdk/keyvault/tests.yml

```diff
@@ -9,3 +9,11 @@ extends:
       # Runs samples with live tests.
       # THIS VARIABLE IS A ONE-OFF WORKAROUND FOR KEYVAULT TESTS SPECIFICALLY, DON'T COPY IT
       AZURE_KEYVAULT_TEST_MODE: Live
+
+    # Managed HSMs can be allocated 1 per subscription per region.
+    # Declare a custom platform to limit test runs to a single HSM.
+    AdditionalPlatforms:
+      Linux_HSM:
+        OSVmImage: "ubuntu-18.04"
+        TestTargetFramework: netcoreapp2.1
+        ArmTemplateParameters: "@{enableHsm = $true}"
```

Comment:
I think you'll need to add one more platform variable:

```
SupportedClouds: 'Public,UsGov'
```

It looks like HSM is not yet available in china and canary regions: [availability](https://azure.microsoft.com/en-us/global-infrastructure/services/?regions=china-non-regional,china-east,china-east-2,china-north,china-north-2,us-east,us-west,us-west-2,non-regional,usgov-non-regional,us-dod-central,us-dod-east,usgov-arizona,usgov-texas,usgov-virginia,us-central,us-east-2,us-north-central,us-south-central,us-west-central&products=azure-dedicated-hsm)

![image](https://user-images.githubusercontent.com/1020379/104501258-77a5f480-55ad-11eb-81d5-84d91d626a63.png)

Reply:
It's in canary, but has a different set of regions, prefixes, and other goop. The two clouds should suffice, though I wonder - since it's just compat testing our KeyClient with Managed HSM (APIs between KV and HSM are designed to be the same), do we need another cloud? HSMs are pretty pricey and limited allocations make them potentially problematic.

Reply:
I think I see now: without this, the pipeline will try all clouds. This limits it. Change made.

Reply:
Yeah, I think it makes sense to just test public here too, given what you say about the intent of the test. In that case just `SupportedClouds: Public`